### PR TITLE
feat(FN-958): Display service options page

### PR DESCRIPTION
**Pull request by francescastocco**
## Introduction
We need to a page to show users a list of services available to them.

## Resolution

- Added the service options page at /service-options.
- Once users login they will be re-directed to this page.
- Checker/Maker/Admin/Read-only users will see link to the portal.
- Payment Officer users will see link to GEF reporting.
- Added E2E tests for this functionality

image
image

### CODE REVIEW

File: e2e-tests/portal/cypress/e2e/journeys/service-options/service-options.spec.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+const { serviceOptions } = require('../../pages');
+const MOCK_USERS = require('../../../fixtures/users');
+const relativeURL = require('../../relativeURL');
+
+const {
+  BANK1_MAKER1, BANK1_CHECKER1, BANK1_PAYMENT_OFFICER1, ADMINNOMAKER, BANK1_READ_ONLY1,
+} = MOCK_USERS;
+
+context('List service options', () => {
```
Comment (ChristianMcc) [inline e2e-tests/portal/cypress/e2e/journeys/service-options/service-options.spec.js:9]: Can we add maker + payment officer seeing both as a test case please?

File: portal/server/routes/service-options.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const express = require('express');
+const { validateToken } = require('./middleware');
+const { getServiceOptions } = require('../controllers/service-options');
+
+const router = express.Router();
+
+router.get('/', validateToken, (req, res) => res.redirect('/service-options'));
```
Comment (ChristianMcc) [inline portal/server/routes/service-options.js:7]: Do you need to delete the equivalent of this in server/routes/dashboard.js line 16?

**Review by ChristianMcc — CHANGES_REQUESTED** [review]
Nice one, some tests are failing as they expect a redirect to deals/dashboard in the API tests instead of /service-options that need to be fixed :)

File: portal/templates/service-options/service-options.njk [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{% extends "service-options/index.njk" %}
+
+{% block pageTitle %}Service Options{% endblock %}
+
+{% block content %}
+  <h1 class="govuk-heading-xl" data-cy="main-heading">What do you want to do?</h1>
```
Comment (ChristianMcc) [inline portal/templates/service-options/service-options.njk:6]: I know we have E2E tests that do the same thing but can we also add a couple of quick component tests for rendering this page with the canAccess variables as well please?

File: portal/server/routes/service-options.js (hunk 2 above)
Comment (francescastocco) [reply portal/server/routes/service-options.js:7]: Yes, I think it is deleted, it's just the test cases which need updating

File: portal/templates/service-options/index.njk [hunk 4]
```diff
@@ -0,0 +1,50 @@
+<!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (francescastocco) [reply portal/templates/service-options/index.njk:1]: [Created](https://ukef-dtfs.atlassian.net/browse/FN-1110)

**Review by ChristianMcc — APPROVED** [review]
LGTM! Can you re-run gef e2e and see if you can get them all passing before merging please

**Comment by francescastocco** [comment]
> LGTM! Can you re-run gef e2e and see if you can get them all passing before merging please

Tests are passing 🎉 please can you approve/ resolve conversations?

**Review by ChristianMcc — DISMISSED** [review]

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**